build-logic-commons/gradle-plugin/src/main/kotlin/gradlebuild.detekt.gradle.kts

```diff
@@ -15,24 +15,28 @@
  */
 
 plugins {
-    id("org.gradle.kotlin-dsl.ktlint-convention")
+    id("io.gitlab.arturbosch.detekt")
 }
 
-tasks {
-    runKtlintCheckOverKotlinScripts {
-        // Only check the build files, not all *.kts files in the project
-        includes += listOf("*.gradle.kts")
-    }
+detekt {
+    // enable all default rules
+    buildUponDefaultConfig = true
 
-    withType<Test>().configureEach {
-        shouldRunAfter(ktlintCheck)
-    }
+    // customize some of the rules, until we can fix the offending cases
+    config.setFrom(project.isolated.rootProject.projectDirectory.file("detekt.yml"))
+
+    // also check .gradle.kts build files
+    val buildFiles = project
+        .fileTree(project.projectDir)
+        .filter { it.name.endsWith("gradle.kts") }
+    source.from(buildFiles)
```

Comment:
❓ This would traverse all project directories including their `build` directory. Doesn't sound like something we'd want. I expect `.gradle.kts` files from source sets are already checked without this so we could simpy use `project.buildFile`. Now would remain the `gradle/shared-with-buildSrc/mirrors.settings.gradle.kts` file but I believe it wasn't checked before so we can ignore it for now.